src/basic/locale-util.c

```diff
@@ -270,6 +272,94 @@ bool is_locale_utf8(void) {
         return (bool) cached_answer;
 }
 
+static Set *keymaps = NULL;
```

Comment:
moving this stuff to locale-util.c is definitely the right approach, but by doing this it becomes "library" code, i.e. no longer code in "main programs". That means we need to be more careful about the context things are run under. In particular, we should really avoid global variables, like this one.

Or to say this differently: library functions should be thread-safe if possible, and this means, no static variables really.